Minerals.gd

```diff
@@ -0,0 +1,75 @@
+extends Spatial
+
+
+# Declare member variables here. Examples:
+enum {
+	NODE,
+	PASSED_TIME,
+	RESPAWN_TIME,
+	STATE
+}
+
+var scene = null
+var mineral = null
+var listBroken = [] # [[mineral, 0, 30, false]] in it 
+var thread
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	scene = get_tree().get_root()
+	loadStateMinerals()
+	thread = Thread.new()
+	thread.start(self, "respawnMinerals", listBroken)
+
+# warning-ignore:shadowed_variable
+func respawnMinerals(listBroken):
+	while (true):
+		for i in range(0, listBroken.size()):
+			if (listBroken[i][STATE] == true
+				and OS.get_ticks_msec() - listBroken[i][PASSED_TIME] >= listBroken[i][RESPAWN_TIME]
+				and is_instance_valid(listBroken[i][NODE])):
+					listBroken[i][NODE].visible = true
```

Comment:
Since you're not removing the node from the queue when the mineral is destroyed, there is no reason to use a thread because the Engine will call the `process` function of each existing mineral event if there are not visible